From 0bc7df5206fee28af956a494dbecb0f3893d1f03 Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@gmail.com>
Date: Wed, 29 Apr 2015 15:46:57 -0700
Subject: [PATCH 0567/1164] CDH-39329 : HIVE-8165 : Annotation changes for
 replication (Sushanth Sowmyan, reviewed by Alan
 Gates)

(cherry picked from commit 0cad50a193ba777f9271808f057caae674738817)

Change-Id: Ia6b75a37bc70b594a11e666598200097b664929e
---
 .../org/apache/hive/hcatalog/api/HCatClient.java   |    5 +++++
 .../hive/hcatalog/api/repl/ReplicationTask.java    |    3 +++
 .../hadoop/hive/metastore/HiveMetaStoreClient.java |    4 ++++
 .../hadoop/hive/metastore/IMetaStoreClient.java    |    6 ++++++
 4 files changed, 18 insertions(+)

diff --git a/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/HCatClient.java b/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/HCatClient.java
index 45b9a2f..c932d85 100644
--- a/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/HCatClient.java
+++ b/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/HCatClient.java
@@ -496,6 +496,7 @@ public abstract void cancelDelegationToken(String tokenStrForm)
    * @return an iterator over a list of replication events that can be processed one by one.
    * @throws HCatException
    */
+  @InterfaceStability.Evolving
   public abstract Iterator<ReplicationTask> getReplicationTasks(
       long lastEventId, int maxEvents, String dbName, String tableName) throws HCatException;
 
@@ -512,6 +513,8 @@ public abstract void cancelDelegationToken(String tokenStrForm)
    * the order that the operations were done on the database.
    * @throws HCatException
    */
+  @InterfaceAudience.LimitedPrivate({"Hive"})
+  @InterfaceStability.Evolving
   public abstract List<HCatNotificationEvent> getNextNotification(long lastEventId,
                                                                   int maxEvents,
                                                                   IMetaStoreClient.NotificationFilter filter)
@@ -522,6 +525,8 @@ public abstract void cancelDelegationToken(String tokenStrForm)
    * @return
    * @throws HCatException
    */
+  @InterfaceAudience.LimitedPrivate({"Hive"})
+  @InterfaceStability.Evolving
   public abstract long getCurrentNotificationEventId() throws HCatException;
 
   /**
diff --git a/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/repl/ReplicationTask.java b/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/repl/ReplicationTask.java
index 47600f7..c5a3028 100644
--- a/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/repl/ReplicationTask.java
+++ b/hcatalog/webhcat/java-client/src/main/java/org/apache/hive/hcatalog/api/repl/ReplicationTask.java
@@ -19,6 +19,8 @@
 package org.apache.hive.hcatalog.api.repl;
 
 import com.google.common.base.Function;
+import org.apache.hadoop.hive.common.classification.InterfaceStability;
+
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hive.hcatalog.api.HCatClient;
 import org.apache.hive.hcatalog.api.HCatNotificationEvent;
@@ -29,6 +31,7 @@
  * ReplicationTask captures the concept of what it'd take to replicate changes from
  * one warehouse to another given a notification event that captures what changed.
  */
+@InterfaceStability.Evolving
 public abstract class ReplicationTask {
   protected HCatNotificationEvent event;
   protected StagingDirectoryProvider srcStagingDirProvider = null;
diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
index 5b626dc..d564434 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
@@ -48,6 +48,7 @@
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.hive.common.ObjectPair;
 import org.apache.hadoop.hive.common.ValidTxnList;
+import org.apache.hadoop.hive.common.classification.InterfaceAudience;
 import org.apache.hadoop.hive.common.classification.InterfaceAudience.Public;
 import org.apache.hadoop.hive.common.classification.InterfaceStability.Unstable;
 import org.apache.hadoop.hive.conf.HiveConf;
@@ -1944,6 +1945,7 @@ public ShowCompactResponse showCompactions() throws TException {
     return client.show_compact(new ShowCompactRequest());
   }
 
+  @InterfaceAudience.LimitedPrivate({"HCatalog"})
   @Override
   public NotificationEventResponse getNextNotification(long lastEventId, int maxEvents,
                                                        NotificationFilter filter) throws TException {
@@ -1964,11 +1966,13 @@ public NotificationEventResponse getNextNotification(long lastEventId, int maxEv
     }
   }
 
+  @InterfaceAudience.LimitedPrivate({"HCatalog"})
   @Override
   public CurrentNotificationEventId getCurrentNotificationEventId() throws TException {
     return client.get_current_notificationEventId();
   }
 
+  @InterfaceAudience.LimitedPrivate({"Apache Hive, HCatalog"})
   @Override
   public FireEventResponse fireListenerEvent(FireEventRequest rqst) throws TException {
     return client.fire_listener_event(rqst);
diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java b/metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java
index 653cc3d..f434f31 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java
@@ -19,6 +19,7 @@
 package org.apache.hadoop.hive.metastore;
 
 import org.apache.hadoop.hive.common.ValidTxnList;
+import org.apache.hadoop.hive.common.classification.InterfaceAudience;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.metastore.api.CompactionType;
 import org.apache.hadoop.hive.metastore.api.CurrentNotificationEventId;
@@ -1382,6 +1383,7 @@ void compact(String dbname, String tableName, String partitionName,  CompactionT
    * A filter provided by the client that determines if a given notification event should be
    * returned.
    */
+  @InterfaceAudience.LimitedPrivate({"HCatalog"})
   interface NotificationFilter {
     /**
      * Whether a notification event should be accepted
@@ -1403,6 +1405,7 @@ void compact(String dbname, String tableName, String partitionName,  CompactionT
    * the order that the operations were done on the database.
    * @throws TException
    */
+  @InterfaceAudience.LimitedPrivate({"HCatalog"})
   NotificationEventResponse getNextNotification(long lastEventId, int maxEvents,
                                                 NotificationFilter filter) throws TException;
 
@@ -1411,6 +1414,7 @@ NotificationEventResponse getNextNotification(long lastEventId, int maxEvents,
    * @return last used id
    * @throws TException
    */
+  @InterfaceAudience.LimitedPrivate({"HCatalog"})
   CurrentNotificationEventId getCurrentNotificationEventId() throws TException;
 
   /**
@@ -1420,6 +1424,8 @@ NotificationEventResponse getNextNotification(long lastEventId, int maxEvents,
    * @return response, type depends on type of request
    * @throws TException
    */
+
+  @InterfaceAudience.LimitedPrivate({"Apache Hive, HCatalog"})
   FireEventResponse fireListenerEvent(FireEventRequest request) throws TException;
 
   class IncompatibleMetastoreException extends MetaException {
-- 
1.7.9.5

